### ISSUE TITLE
Feedback

### ISSUE THREAD

**Issue by esin87** [income3000/blueGang#1]
# Project 2 Feedback

## Code Quality - Progressing

Criteria: Is the code well-formatted? Are variable and function names semantic and sensible? Is the code easy to read and understand?

### Strengths:

- Code with semantic and sensible variable/function/component naming throughout.
- Code is easy to read and understand. Components are named and organized in a way that makes sense and is straightforward to follow.
- Great job breaking down UI components into individual React components. As a result, component files are relatively concise.

### Areas for Growth:

- Code is not formatted consistently. Be sure indents and spacing are uniform throughout your application. You can configure your text editor to do this for you automatically!
- There are still instances of console.logs and commented-out code that should be removed from the deployed main branch. Check the console in development and try to eliminate the errors and warnings that you see there.
- Be sure to delete any unused files and imports that come with the create-react-app defaults, such as `App.test.js`, other unused test files, and the React logos. Also be sure to update the `<title>` of `index.html` and the `favicon` with the relevant info for your app. Also be sure to delete any empty files that you decided you don't need, such as `Router.js`.
- You pushed your `node_modules` to your GitHub repo and appear to have lost your `.gitignore` file. I'd add a `.gitignore` file back to the root of your repo and add the standard CRA contents to it. To remove your `node_modules` from your Git cache you can run the following:

```bash
git rm -r --cached .
git add .
git commit -m "remove gitignore files"
```

...then push it back up to your repo.

## Technical Requirements - Performing

Criteria: How does the project stack up to the requirements for this project? Is the developer making use of the material we've covered in a way that makes sense?

### Strengths:

- The project meets the technical requirements laid out for this project.
- Nice job implementing React Router for page navigation and multi-page functionality in a SPA.

### Areas for Growth:

- You have `<Routes>` in both `index.js` and `App.js` which appears to be causing your `About` page to load without the header and outside of the context of the `App` component.
- You have both a `yarn.lock` and `package-lock.json`, which indicates that you've used both `yarn` and `npm` as your package managers -- choose one or the other! It can cause problems down the line if we use both in a given application.
- Your GitHub repo shows a good history of commits, but work on both the consistency of your commit message formatting as well as their content. Ideally, you should have one change per commit. The standard format is usually 'capitalized present-tense verb + description of what was done' for each and every commit. For example, 'Add styling to card container' or 'Refactor API call'. This is my favorite [article](https://chris.beams.io/posts/git-commit/) on how to write good commit messages and why it should be a focus for developers.

## Creativity and Interface - Progressing

Criteria: Is the application easy to navigate? Does it work well in every major browser? Is it responsive? Does it incorporate modern UI Themes?

### Strengths:

- There is some basic styling in place!

### Areas for Growth

- I know you had a challenging week in terms of hardware issues. I'm glad you prioritized functionality over styling, but at some point you'll want to revisit the design of the application to make it look more polished, modern, and responsive to browser sizes!

## Functionality - Performing

Criteria: Does the application work without errors or bugs? Does it present a complete app, where every feature is fully implemented in a way that makes sense?

### Strengths:

- Application works without any major errors or bugs observed. No errors related to code you wrote log to the console.
- Project presents a complete application.

### Areas for Growth:

- The search functionality is slightly buggy! I would refactor the code so that the search is submitted after the user is done typing their search string (`onSubmit`) vs. as they type into the form input element (`onChange`).
- You'd also want to introduce some type of error handling so that bad search strings or failed API calls result in some type of user feedback. Alternatively, you could use a `<select>` element and pre-fill the search options since users can only search by state.
- Some type of "Loading" text or component might be good user feedback while you're waiting on the results of the API call to return.

## Presentation - Performing

Criteria: Is there adequate documentation? Is the repository well-organized and free of clutter?

### Strengths:

- The repository is well-organized and free of clutter. Files are well-named.
- There is a start to the documentation in your README that does a good job of highlighting your application's purpose.

### Areas for Growth:

- I would organize the component files into their own `components` folder to make the `src` folder a little easier to sort through.
- Your README should include information about your coding and planning process, as it's the first point of entry to someone coming into contact with your code. The project 2 markdown specifies that it should include at least the following areas: Description (which you have), Technologies Used, Installation Instructions, and Contribution Guidelines.

## Hard Requirements - (Mostly) Complete:

- [x] Your app must be a working, interactive, React application.
- [x] Your app must use React Router and have at least one additional page view.
- [x] It must include data from a third-party API.
- [x] It must use at least 4 separate components.
- [x] It must be deployed to the Web.
- [x] It uses CSS Grid or Flex for layout and not floats.
- [ ] Your project repository must adequately document your project and include a line to the deployed app. ❌ - this is a quick fix!
- [x] You have used git for version control throughout the project.

## Grade: Pass 🎉🎉🎉

Louis, good work on this project! I know you had a challenging week in terms of multiple hardware issues that had a big impact on how much time you could devote to coding. Nevertheless, you delivered an MVP that meets the technical goals of the assignment, and as you say in the markdown, brings attention to a huge issue in our country. Now that you've hit MVP, it would be great to keep building out the features, design, and functionality of this application so that you can meet your stretch goals and further develop this important project. Remember that the README is an important part of the project as well -- yours should include some additional information that highlights your planning and development process. Congrats on completing this project, and thank you for all your hard work and engagement in the course. Looking forward to seeing your continued growth!
